hq/app/logic/IamUnrecognisedUsers.scala

```diff
@@ -49,10 +47,13 @@ object IamUnrecognisedUsers extends Logging {
     accountCredsReports: List[(AwsAccount, CredentialReportDisplay)],
     janusUsernames: List[String],
     allowedAccountIds: List[String]
-  ): List[(AwsAccount, List[HumanUser])] = {
-    val unrecognisedUsers = accountCredsReports.map { case (acc, crd) => (acc, filterUnrecognisedIamUsers(crd.humanUsers, janusUsernames)) }
-    val unrecognisedUsersInAllowedAccounts = filterAllowedAccounts(unrecognisedUsers, allowedAccountIds)
-    unrecognisedUsersInAllowedAccounts.filter { case (_, users) => users.nonEmpty } // remove accounts with an empty list of users
+  ): List[AccountUnrecognisedUsers] = {
+    accountCredsReports.map {
+      case (acc, crd) => AccountUnrecognisedUsers(acc, filterUnrecognisedIamUsers(crd.humanUsers, janusUsernames))
+    }.filter { accountUsers =>
+      accountUsers.unrecognisedUsers.nonEmpty && // remove accounts with an empty list of users
+        allowedAccountIds.contains(accountUsers.account.id)
+    }
```

Comment:
Changed the return type to our convenience type `AccountUnrecognisedUsers` instead of a list of tuples. Also inlines the filtering logic to remove the unnecessary misdirection of `filterAllowedAccounts`.